cpp/src/arrow/dataset/file_base.h

```diff
@@ -211,8 +221,12 @@ class ARROW_DS_EXPORT FileFragment : public Fragment,
 
   FileSource source_;
   std::shared_ptr<FileFormat> format_;
-
   friend class FileFormat;
+
+  // we do not want ParquetFileFragment to inherit these things.
```

Comment:
Er... if we do not expect inheritance, we should probably put in a derived class?

Reply:
Yeah.... This probably will change anyways. The thing I am going to do after this one gets merged is to incorporate @zhztheplayer's parquet slicing using this API -- so perhaps we do expect inheritance. And it's fine I think if FileFormats who currently don't support slicing get those attributes as long as they are private and can't be accessed.